src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -0,0 +1,117 @@
+/* global Promise */
+import { _, $ } from 'okta';
+import BaseView from '../internals/BaseView';
+import BaseForm from '../internals/BaseForm';
+
+const request = (url, method = 'GET', body = null, timeout = 1000) => {
+  return $.ajax({
+    url,
+    method,
+    timeout,
+    contentType: 'application/json',
+    data: body,
+  });
+};
+
+const Body = BaseForm.extend({
+  noButtonBar: true,
+
+  className: 'ion-form device-challenge-poll',
+
+  initialize () {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    this.deviceChallengePollRemediation = this.options.appState.get('remediation').find(
+      rem => rem.name === 'device-challenge-poll'
+    );
+    this.doChallenge();
+    this.startPolling();
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    this.stopPolling();
+  },
+
+  doChallenge () {
+    const deviceChallengeHintsKey = this.deviceChallengePollRemediation.relatesTo;
+    const deviceChallenge = deviceChallengeHintsKey &&
+      this.options.appState.get('currentState')[deviceChallengeHintsKey];
+    switch (deviceChallenge.challengeMethod) {
+    case 'LOOPBACK':
+      this.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+      break;
+    case 'custom-scheme':
+      this.doCustomURI(deviceChallenge.href);
+      break;
+    }
+  },
+
+  doLoopback (authenticatorDomainUrl = '', ports = [], challengeRequest = '') {
+    let currentPort;
+    let foundPort = false;
+
+    const getAuthenticatorUrl = (path) => {
+      // To use the mock server by running `yarn authenticator`
+      // return `http://localhost:4000/${path}?authenticator=${currentPort}`;
+
+      if (authenticatorDomainUrl === 'localhost:3000') {
```

Comment:
this doesn't seem ideal. 
I think we shall leave out this specific logic for local testing and find another way to make testing easier because for now
- in testcafe, you shall be able to mock response per port. port is just part of URL. (let me know if you had tried but didn't work)
- in mock server, you could just mock one port, e.g. `6512` hence given port list `[ "2000", "6511", "6512", "6513" ],`, the first 2 will be error and third port probing will succeed.

Reply:
I agree it is not ideal. But this is so far the only way I can make the test work. When the domain is different, even though Cross domain header is set in the response in RequestMock, the browser keeps throwing ERR_EMPTY_RESPONSE error. 
https://[a different domain than localhost]:[a different port than 3000] does not work
https://localhost:[a different port than 3000] does not work
https://localhost:3000/xxx works with cross domain header enabled

Reply:
I'm guessing it's because you set ` "domain": "localhost:3000"` in the mock response.